File: Examples/VisualBasic/ModifyingAndConvertingImages/JPEG/CroppingByShifts.vb
'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.JPEG
    Class CroppingByShifts
        Public Shared Sub Run()
            ' ExStart:CroppingByShifts
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_JPEG()

            ' Load an existing image into an instance of RasterImage class
            Using rasterImage As RasterImage = DirectCast(Image.Load(dataDir & Convert.ToString("aspose-logo.jpg")), RasterImage)
                ' Before cropping, the image should be cached for better performance
                If Not rasterImage.IsCached Then
                    rasterImage.CacheData()
                End If

                ' Define shift values for all four sides
                Dim leftShift As Integer = 10
                Dim rightShift As Integer = 10
                Dim topShift As Integer = 10
                Dim bottomShift As Integer = 10

                ' Based on the shift values, apply the cropping on image
                ' Crop method will shift the image bounds toward the center of image
                rasterImage.Crop(leftShift, rightShift, topShift, bottomShift)

                ' Save the results to disk
                rasterImage.Save(dataDir & Convert.ToString("CroppingByShifts_out.jpg"))
            End Using
        End Sub
        ' ExEnd:CroppingByShifts
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/JPEG/CroppingByShifts.cs
/*
This project uses the automatic package‑restore feature of NuGet to resolve the Aspose.Imaging for .NET API
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us using https://forum.aspose.com/
*/

using System;

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.JPEG
{
    class CroppingByShifts
    {
        public static void Run()
        {
            //ExStart:CroppingByShifts
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JPEG();

            Console.WriteLine("Running example CroppingByShifts");
            // Load an existing image into an instance of the RasterImage class.
            using (RasterImage rasterImage = (RasterImage)Image.Load(dataDir + "aspose-logo.jpg"))
            {
                // Before cropping, cache the image for better performance.
                if (!rasterImage.IsCached)
                {
                    rasterImage.CacheData();
                }

                // Define shift values for all four sides.
                int leftShift = 10;
                int rightShift = 10;
                int topShift = 10;
                int bottomShift = 10;

                // Apply cropping based on the shift values. The Crop method shifts the image bounds toward
                // the center of the image, and the result is saved to disk.
                rasterImage.Crop(leftShift, rightShift, topShift, bottomShift);
                rasterImage.Save(dataDir + "CroppingByShifts_out.jpg");
            }

            Console.WriteLine("Running example CroppingByShifts");
            //ExEnd:CroppingByShifts
        }        
    }
}